### PR TITLE
Update the min Go version for building from source

### PR DESCRIPTION
**Pull request by jseldess**
Release note: None

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -35,7 +35,7 @@ new engineers.
      - On GNU/Linux, the terminfo development libraries, which may be
        part of a ncurses development package (e.g. `libncurses-dev` on
        Debian/Ubuntu, but `ncurses-devel` on CentOS).
-     - A Go 1.9+ environment with a recent 64-bit version of the toolchain. Note that
+     - A Go 1.14+ environment with a recent 64-bit version of the toolchain. Note that
```
Comment (petermattis) [inline CONTRIBUTING.md]: Go 1.11.4.

Comment (jseldess) [reply CONTRIBUTING.md]: Fixed. Sorry about that.

**Comment by jseldess** [comment]
@petermattis, all good now?

**Review by petermattis — APPROVED** [review]

**Pull request merged**